kolibri/core/assets/src/vue/assessment-wrapper/index.vue

```diff
@@ -0,0 +1,132 @@
+<!--
+This template is intended to act as the default wrapper for assessment focused rendering
+plugins.
+
+As such, it provides display of current mastery progress, and manages all mastery/attempt log
+oriented data synchronization.
+-->
+<template>
+
+  <div>
+    <slot v-if="ready"></slot>
+  </div>
+
+</template>
+
+
+<script>
+
+  const logging = require('kolibri/lib/logging').getLogger(__filename);
+  const actions = require('kolibri/coreVue/vuex/actions');
+  const hint = require('../../constants').InteractionTypes.hint;
+
+  module.exports = {
+    props: {
+      itemId: {
+        type: String,
+      },
+      masteryModel: {
+        type: String,
+      },
+      masterySpacingTime: {
+        type: Number,
+      },
+      masteryCriterion: {
+        type: String,
+      }
+    },
+    data: () => ({
+      ready: false,
+    }),
+    created() {
+      this.$on('checkanswer', (correct, complete, firstAttempt, hinted) => { this.updateMasetryLogSaveAttemptLog(correct, complete, firstAttempt, hinted);});
+      this.$on('takehint', (firstAttempt, hinted) => { this.hintTaken(firstAttempt, hinted);});
+      this.$on('passexercise', () => { this.exercisePassed();});
```

Comment:
I see here why you used `this.$parent.$emit` in the perseus renderer.

I think more appropriate would be for perseus to say `this.$emit` and then in the html above you can do:

```html
<slot
  v-if="ready"
  @checkanswer="somemethod"
  @takehint="somemethod"
  @passexercise="somemethod"
></slot>
```